app/init.php

```diff
@@ -1180,3 +1185,240 @@ function getDevice($root): Device
     }
     return $requestTimestamp;
 }, ['request']);
+
+App::setResource('eventHandler', function(App $utopia, ?Logger $logger, array $loggerBreadcrumbs, Request $request, Document $project, Document $user, Event $events) {
+    return function (string $databaseEvent, Document $document) use ($utopia, $logger, $loggerBreadcrumbs, $request, $project, $user, $events) {
+        $route = $utopia->match($request);
+        $response = new Response(new SwooleResponse());
+        try {
+            $dump = function($value) {
+                $p = var_export($value, true);
+                $b = debug_backtrace();
+                print($b[0]['file'] . ':' . $b[0]['line'] . ' - ' . $p . "\n");
+            };
+            $dump('in ' . $route->getMethod() . ' ' . $route->getPath());
+
+            $queue = '';
+            $class = '';
+            $event = new Event('', '');
+
+            // $event
+            // ->setContext('collection', $collection)
+            // ->setContext('database', $db);
+
+            $event
+                ->setProject($project)
+                ->setUser($user);
+                
+            $eventPattern = $route->getLabel('event', '');
+            $eventPatternReady = true;
+            $modelType = $route->getLabel('sdk.response.model', '');
+            $collection = $document->getCollection();
+            $dump($databaseEvent);
+            $dump($collection);
+
+            switch ($collection) {
+                case 'users':
+                    if (!\str_starts_with($eventPattern, 'users.[userId]')) {
+                        $eventPattern = 'users.[userId]';
+                        $eventPatternReady = false;
+                    }
```

Comment:
reverse engineering the event pattern given the collection feels a little hacky.